File: src/main/java/ir/reservs/reservs/ui/main/information/InformationContract.java

package ir.reservs.reservs.ui.main.information;

import ir.reservs.reservs.ui.base.BaseFragmentContract;

public class InformationContract {
    interface View extends BaseFragmentContract.View {
        void nameUpdated(String name);

        void initializeViews(String name);
    }

    interface Presenter extends BaseFragmentContract.Presenter<View> {
        void confirmInformation(String name);

        void initializeViews();
    }
}

File: src/main/java/ir/reservs/reservs/ui/main/information/InformationContract.kt
package ir.reservs.reservs.ui.main.information

import androidx.fragment.app.FragmentManager
import ir.reservs.reservs.ui.base.BaseFragmentContract

class InformationContract {
    interface View : BaseFragmentContract.View {
        fun informationUpdated()
        fun displayDate(date: String)
        fun getFManager(): FragmentManager;
        fun initializeViews(name: String, age: String)
    }

    interface Presenter : BaseFragmentContract.Presenter<View>
}